Clarify strict validation of examples (#734)
raml-org/raml-spec: versions/raml-10/raml-10.md

### Defining Examples in RAML

> It is highly RECOMMENDED that API documentation include a rich selection of examples. RAML supports either the definition of multiple examples or a single one for any given instance of a type declaration. In addition to YAML by default, processors SHOULD support JSON and XML representations of examples. Processors MAY support additional formats. Note that type definition is agnostic to example encoding such that, e.g. examples in YAML will work for JSON or XML, and in reverse for any chosen combination of those three supported encodings.

Reviewer: ```suggestion
It is highly recommended that the API documentation includes a rich selection of examples. RAML supports either the definition of multiple examples or a single one for any given instance of a type declaration. In addition to YAML(default), processors must support JSON and XML examples. Processors may support additional formats. Note that the type definition is agnostic to example encoding such that the examples in YAML work for JSON or XML, and in reverse for any chosen combination of those three supported encodings.
```

> It is highly RECOMMENDED that API documentation include a rich selection of examples. RAML supports either the definition of multiple examples or a single one for any given instance of a type declaration. In addition to YAML by default, processors SHOULD support JSON and XML representations of examples. Processors MAY support additional formats. Note that type definition is agnostic to example encoding such that, e.g. examples in YAML will work for JSON or XML, and in reverse for any chosen combination of those three supported encodings.

Reviewer 2: Question, please clarify: What is mean by "and in reverse"? I think it probably means that you can write the example in any of the three formats will work for the others, but I'm not 100% sure if that's correct. Can you clarify that? I did not attempt to modify that part of the paragraph.

```suggestion
It is highly RECOMMENDED that API documentation include a rich selection of examples. RAML supports either the definition of multiple examples or a single one for any given instance of a type declaration. In addition to supporting YAML by default, processors SHOULD support JSON and XML representations of examples. Processors MAY support additional formats. Note that type definition is agnostic to example encoding; for example, examples in YAML will work for JSON or XML, and in reverse for any chosen combination of those three supported encodings.
```

Author: I think it means "vice versa".

Reviewer: ```suggestion
It is highly RECOMMENDED that the API documentation includes a rich selection of examples. RAML supports either the definition of multiple examples or a single one for any given instance of a type declaration. In addition to YAML(default), processors SHOULD support JSON and XML representations of examples. Processors MAY support additional formats. Note that type definition is agnostic to example encoding, for example, examples in YAML will work for JSON or XML, and vice versa, for any chosen combination of those three supported encodings.
```

Reviewer: or you can put it like this:
Ensure that your API documentation includes a rich selection of examples. RAML supports the definition of either multiple examples or a single one for any given instance of a type declaration. In addition to supporting YAML by default, processors SHOULD support JSON and XML representations of examples, and OPTIONALLY support additional formats. Note that type definition is agnostic to example encoding; examples in any of YAML, JSON, or XML also work for the other two.

Author: I think I prefer `In addition to supporting YAML by default`. 

Regarding the last sentence I prefer the: 

> Note that type definition is agnostic to example encoding, for example, examples in YAML will work for JSON or XML, and vice versa, for any chosen combination of those three supported encodings.

form, although I am not sure we need the `, for example,` part. Can we say: 

> Note that type definition is agnostic to example encoding: examples in YAML will work for JSON or XML, and vice versa, for any chosen combination of those three supported encodings.

instead?

Reviewer: sure

> It is highly RECOMMENDED that API documentation include a rich selection of examples. RAML supports either the definition of multiple examples or a single one for any given instance of a type declaration. In addition to YAML by default, processors SHOULD support JSON and XML representations of examples. Processors MAY support additional formats. Note that type definition is agnostic to example encoding such that, e.g. examples in YAML will work for JSON or XML, and in reverse for any chosen combination of those three supported encodings.

Reviewer 2: ```suggestion
It is highly RECOMMENDED that API documentation include a rich selection of examples. RAML supports either the definition of multiple examples or a single one for any given instance of a type declaration. In addition to YAML by default, processors SHOULD support JSON and XML representations of examples. Processors MAY support additional formats. Note that type definition is agnostic to example encoding; for example, examples in YAML will work for JSON or XML, and in reverse for any chosen combination of those three supported encodings.
```

| description? | A substantial, human-friendly description for an example. Its value is a string and MAY be formatted using [markdown](#markdown).
| (&lt;annotationName&gt;)? | [Annotations](#annotations) to be applied to this API. An annotation is a map having a key that begins with "(" and ends with ")" where the text enclosed in parentheses is the annotation name, and the value is an instance of that annotation.
| value | The actual example of a type instance.

> | strict? | Validates the example against any type declaration (the default). Set this to false to prevent validation.

Reviewer: ```suggestion
| strict? | Validates the example against any type declaration (default). Set this to false to prevent validation.
```
